helper/common/shared_state.go

```diff
@@ -15,14 +19,67 @@ func sharedStateFilename(suffix string, buildName string) string {
 }
 
 func SetSharedState(key string, value string, buildName string) error {
-	return ioutil.WriteFile(sharedStateFilename(key, buildName), []byte(value), 0600)
+	uuid := os.Getenv("PACKER_RUN_UUID")
+
+	// Encrypt the value using the run uuid. This is probably good enough as an
+	// encryption key because we only keep it in memory until after the point
+	// at which this storage file is wiped.
+	encryptionKey := []byte(uuid)[0:32]
```

Comment:
this key is a bit weak since it's based on 16 bits of data (see https://github.com/hashicorp/uuid/blob/ebb0a03e909c9c642a36d2527729104324c44fdb/uuid.go#L10).

It's also not cryptographically secure data, since it's from `rand` and not `crypto/rand`.